angr/sim_state.py

```diff
@@ -374,20 +361,20 @@ def _set_plugin_state(self, plugin, inhibit_init=False):
     # Java support
     #
 
-    @property
-    def _is_java_project(self):
-        """
-        Indicates if the project's main binary is a Java Archive.
-        """
-        return self.project and self.project.is_java_project
+    def __getattr__(self, a):
```

Comment:
Again, this is too much magic...

Reply:
This magic was already here, and I'd rather fix it under a different PR. This way, at least it doesn't make self.arch account for 5% of our runtime (though it still takes a chunk).

Reply:
I'm moving it to `__init__()` and see if that works.